lib/MusicBrainz/Server/Edit/Genre/Create.pm

```diff
@@ -0,0 +1,77 @@
+package MusicBrainz::Server::Edit::Genre::Create;
+use Moose;
+
+use MusicBrainz::Server::Constants qw( $EDIT_GENRE_CREATE );
+use MusicBrainz::Server::Edit::Types qw( Nullable );
+use MusicBrainz::Server::Entity::Util::JSON qw( to_json_object );
+use MusicBrainz::Server::Translation qw( N_l );
+use aliased 'MusicBrainz::Server::Entity::PartialDate';
+use Moose::Util::TypeConstraints;
+use MooseX::Types::Moose qw( Str );
+use MooseX::Types::Structured qw( Dict Optional );
+
+use aliased 'MusicBrainz::Server::Entity::Genre';
+
+extends 'MusicBrainz::Server::Edit::Generic::Create';
+with 'MusicBrainz::Server::Edit::Role::Preview';
+with 'MusicBrainz::Server::Edit::Genre';
+with 'MusicBrainz::Server::Edit::Role::AlwaysAutoEdit';
+
+sub edit_name { N_l('Add genre') }
+sub edit_type { $EDIT_GENRE_CREATE }
+sub _create_model { 'Genre' }
+sub genre_id { shift->entity_id }
+
+
+has '+data' => (
+    isa => Dict[
+        name       => Str,
+        gid        => Optional[Str],
+        comment    => Nullable[Str],
+    ]
+);
+
+sub foreign_keys
+{
+    my $self = shift;
+    return {
+        Genre       => [ $self->entity_id ],
+    };
+}
+
+sub build_display_data
+{
+    my ($self, $loaded) = @_;
+
+    return {
+        name    => $self->data->{name},
+        comment => $self->data->{comment},
+        genre   => to_json_object((defined($self->entity_id) &&
+            $loaded->{Genre}{ $self->entity_id }) ||
+            Genre->new( name => $self->data->{name} )
+        ),
+    };
+}
+
+sub _insert_hash
+{
+    my ($self, $data) = @_;
+    return $data;
+};
+
```

Comment:
Doesn't seem needed since it's identical to the method from Edit::Generic::Create.

Reply:
Fair enough. I had copied it from Edit::Area::Create which also doesn't seem to need it, so I removed it from there in a separate commit.